### PR TITLE
Remove PostgresJobStore (compatibility layer)

### LINKED ISSUE #187: Remove the PostgresJobStore compatibility layer
PostgresJobStore is deprecated. We want to remove it in the next release. Now is the right time to remove it, as we're changing the connector API (with #185 for example).

### PR DESCRIPTION
**Pull request by elemoine**
This removes the PostgresJobStore compatibility layer.

Closes #187.

### Successful PR Checklist:
- [ ] Tests  **Not relevant**
- [ ] Documentation (optionally: [run spell checking](https://github.com/peopledoc/procrastinate/blob/master/CONTRIBUTING.rst#build-the-documentation)) **Not relevant**
- [ ] Had a good time contributing? (if not, feel free to give some feedback) **Not relevant** :smile:

### CODE REVIEW

**Comment by elemoine** [comment]
> There's also the "job_store" arg of the App

Yep, fixed.

File: procrastinate/app.py [hunk 1]
```diff
@@ -68,18 +67,7 @@ def __init__(
...
-        # Compatibility
-        if job_store:
-            message = (
-                "Use App(connector=procrastinate.PostgresConnector(...)) "
-                "instead of App(job_store=procrastinate.PostgresJobStore())"
-            )
-            logger.warning(f"Deprecation Warning: {message}")
-            warnings.warn(DeprecationWarning(message))
-            connector = job_store
         if not connector:
```
Comment (ewjoachim) [inline procrastinate/app.py:71]: Connector can now be a mandatory argument and this `if` branch can be removed.

Comment (elemoine) [reply procrastinate/app.py:71]: Yeah, I was too fast.

Comment (elemoine) [reply procrastinate/app.py]: Fixed.

**Comment by elemoine** [comment]
We're getting hit by https://github.com/pytest-dev/pytest-asyncio/issues/154.

**Comment by elemoine** [comment]
@ewjoachim pytest-asyncio 0.11.0 broke our tests (see https://github.com/pytest-dev/pytest-asyncio/issues/154). So I added a commit to use `pytest-asyncio<0.11.0` until this is fixed.

**Comment by elemoine** [comment]
> @ewjoachim pytest-asyncio 0.11.0 broke our tests (see [pytest-dev/pytest-asyncio#154](https://github.com/pytest-dev/pytest-asyncio/issues/154)). So I added a commit to use `pytest-asyncio<0.11.0` until this is fixed.

Good news. The problem is fixed with pytest-asyncio 0.12.0, which just came out.

**Comment by ewjoachim** [comment]
But the nice thing is that we don't even need to change anything now.

**Pull request merged**